sdk/core/Azure.Core/src/AzureKeyCredential.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ComponentModel;
+using System.Threading;
+
+namespace Azure.Core
+{
+    /// <summary>
+    /// API key credential used to authenticate to an Azure Service.
+    /// It provides the ability to update the API key without creating a new client.
+    /// </summary>
+    public class AzureKeyCredential
+    {
+        private string _apiKey;
+
+        /// <summary>
+        /// API key used to authenticate to an Azure service.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public string ApiKey
+        {
+            get => Volatile.Read(ref _apiKey);
+            private set => Volatile.Write(ref _apiKey, value);
```

Comment:
Is it allowed to assign `null` to `_key` here?

Reply:
The only way to set value to property is in the `Update` method which has validation for that.
https://github.com/Azure/azure-sdk-for-net/pull/10668/files#diff-dca2442a6e0ae3f87d86879dc5dc2afbR43